src/test/java/tatracker/logic/parser/sort/SortCommandParserTest.java

```diff
@@ -0,0 +1,77 @@
+package tatracker.logic.parser.sort;
+
+import static tatracker.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static tatracker.logic.parser.CommandParserTestUtil.assertParseSuccess;
+
+import org.junit.jupiter.api.Test;
+
+import tatracker.commons.core.Messages;
+import tatracker.logic.commands.sort.SortCommand;
+import tatracker.logic.commands.sort.SortGroupCommand;
+import tatracker.logic.commands.sort.SortModuleCommand;
+import tatracker.logic.commands.sort.SortType;
+
+public class SortCommandParserTest {
+
+    private SortCommandParser parser = new SortCommandParser();
+
+    @Test
+    public void parse_allFieldsPresent_success() {
+        SortType type = SortType.ALPHABETIC;
+        String expectedModule = "CS3243";
+        String expectedGroup = "G04";
+
+
+        // whitespace only preamble
+        assertParseSuccess(parser, " group m/CS3243 g/G04 t/alpha",
```

Comment:
No group isn't a preamble. Group is part of the full command word.
The preamble would be this: ____blanks____m/CS3243
Think of it as sort group + WHITESPACES + arguments